services/121-service/src/programs/program.entity.ts

```diff
@@ -154,4 +163,91 @@ export class ProgramEntity extends CascadeDeleteEntity {
   // This is an array of ProgramQuestionEntity names that build up the full name of a PA.
   @Column('json', { nullable: true })
   public fullnameNamingConvention: JSON;
+
+  public async getValidationInfoForQuestionName(
+    name: string,
+  ): Promise<ValidationInfo> {
+    if (name === Attributes.paymentAmountMultiplier) {
+      return { type: AnswerTypes.numeric };
+    } else if (name === Attributes.phoneNumber) {
+      return { type: AnswerTypes.tel };
+    }
+
+    const repo = getConnection().getRepository(ProgramEntity);
+    const resultProgramQuestion = await repo
+      .createQueryBuilder('program')
+      .leftJoin('program.programQuestions', 'programQuestion')
+      .where('program.id = :programId', { programId: this.id })
+      .andWhere('programQuestion.name = :name', { name: name })
+      .select('"programQuestion"."answerType"', 'type')
+      .addSelect('"programQuestion"."options"', 'options')
+      .getRawOne();
+
+    const resultFspQuestion = await repo
+      .createQueryBuilder('program')
+      .leftJoin('program.financialServiceProviders', 'fsp')
+      .leftJoin('fsp.questions', 'question')
+      .where('program.id = :programId', { programId: this.id })
+      .andWhere('question.name = :name', { name: name })
+      .select('"question"."answerType"', 'type')
+      .addSelect('"question"."options"', 'options')
+      .getRawOne();
+
+    const resultProgramCustomAttribute = await repo
+      .createQueryBuilder('program')
+      .leftJoin('program.programCustomAttributes', 'programCustomAttribute')
+      .where('program.id = :programId', { programId: this.id })
+      .andWhere('programCustomAttribute.name = :name', { name: name })
+      .select('"programCustomAttribute".type', 'type')
+      .getRawOne();
+
+    const repoInstance = getConnection().getRepository(InstanceEntity);
+    const monitoringQuestion = await repoInstance
```

Comment:
make relation with program @RubenGeo

Reply:
Instance is not related to program in the end